addon/src/-private/array.ts

```diff
@@ -106,6 +106,18 @@ class TrackedArray<T = unknown> {
       get(target, prop /*, _receiver */) {
         let index = convertToInt(prop);
 
+        if (prop === 'splice') {
+          return (
+            ...args: [start: number, deleteCount?: number | undefined]
```

Comment:
Without this very precise casting I could not make the line `114` not complain about "A spread argument must either have a tuple type or be passed to a rest parameter.". Maybe there is better solution though.